### ISSUE TITLE
VAE代码和书上的有一点点不一样

### ISSUE THREAD

**Issue by LSC-getup** [dragen1860/Deep-Learning-with-TensorFlow-book#39]
书12.5VAE实战中

本节我们基于 VAE 模型实战 Fashion MNIST 图片的重建与生成。如图 12.12 所示，输
入为 Fashion MNIST 图片向量，经过 3 个全连接层后得到隐向量𝐳的均值与方差，分别用 2
个输出节点数为 20 的全连接层表示，FC2 的 20 个输出节点表示 20 个特征分布的均值向量μ
，FC3 的 20 个输出节点表示 20 个特征分布的方差向量的𝑙𝑜𝑔值。通过 Reparameterization 
trick 采样获得长度为 20 的隐向量𝐳，并通过 FC4/FC5 重建出样本图片。

但是代码里的是
`z_dim = 10

class VAE(keras.Model):

    def __init__(self):
        super(VAE, self).__init__()

        # Encoder
        self.fc1 = layers.Dense(128)
        self.fc2 = layers.Dense(z_dim)  # 均值输出
        self.fc3 = layers.Dense(z_dim)  # 方差输出

        # Decoder
        self.fc4 = layers.Dense(128)
        self.fc5 = layers.Dense(784)`

另外一处在重建图片那里
`
### 重建图片，从测试集采样图片
    x = next(iter(test_db))
    x = tf.reshape(x, [-1, 784])
    x_hat_logits, _, _ = model(x)
    x_hat = tf.sigmoid(x_hat_logits)
    x_hat = tf.reshape(x_hat, [-1, 28, 28]).numpy() * 255.
    x_hat = x_hat.astype(np.uint8)
    save_images(x_hat, 'vae_images/rec_epoch%d.png' % epoch)`

**Comment by dragen1860** [comment]
这个隐变量维度其实比较自由，10,20都可以。书上应该是用的20，你可以将代码里面改为20.谢谢。
